script.js

```diff
@@ -1,4 +1,150 @@
+var userName = "";
+
+// Creating functions to randomly output "scissors", "paper", or "stone"
+
+var rollingRandomNumber = function (input) {
+  //return 2; //for testing; remove when done testing
+  var randomOutput = Math.floor(Math.random() * 3) + 1;
+  console.log("rolling random number to be assigned to SPS later");
+  return randomOutput;
+};
+
+var assignNumberToSPS = function (input) {
+  var rolledNumber = rollingRandomNumber();
+  console.log(
+    "rolling random number to assign to SPS. the number is " + rolledNumber
+  );
+  var numberToSPS = "some number was assigned but it wasnt 1 or 2 or 3";
+
+  //Assign random number to scissors, paper or stone
+
+  if (rolledNumber == 1) {
+    numberToSPS = scissors;
+  }
+
+  if (rolledNumber == 2) {
+    numberToSPS = paper;
+  }
+
+  if (rolledNumber == 3) {
+    numberToSPS = stone;
+  }
+
+  console.log("SPS is " + numberToSPS);
+  return numberToSPS;
+};
+
+// Create functions for win, draw and loss;
+
+var userWins = function (userGuess, programmeOutputsSPS) {
+  return (
+    ((userGuess == scissors || userGuess == reverseStone) &&
+      programmeOutputsSPS == paper) ||
+    ((userGuess == paper || userGuess == reverseScissors) &&
+      programmeOutputsSPS == stone) ||
+    ((userGuess == stone || userGuess == reversePaper) &&
+      programmeOutputsSPS == scissors)
+  );
+};
+
+var userDraws = function (userGuess, programmeOutputsSPS) {
+  return (
+    userGuess == programmeOutputsSPS ||
+    (userGuess == reversePaper && programmeOutputsSPS == paper) ||
+    (userGuess == reverseStone && programmeOutputsSPS == stone) ||
+    (userGuess == reverseScissors && programmeOutputsSPS == scissors)
+  );
+};
+
+var userLoses = function (userGuess, programmeOutputsSPS) {
+  return (
+    ((userGuess == scissors || userGuess == reversePaper) &&
+      programmeOutputsSPS == stone) ||
+    ((userGuess == paper || userGuess == reverseStone) &&
+      programmeOutputsSPS == scissors) ||
+    ((userGuess == stone || userGuess == reverseScissors) &&
+      programmeOutputsSPS == paper)
+  );
+};
+
+// Create main function:
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var programmeOutputsSPS = assignNumberToSPS();
+  console.log("programe outputs " + programmeOutputsSPS);
+
+  // If game mode is "waiting on user name" ...
+
+  if (gameMode == "waiting for username") {
+    userName = input;
+    gameMode = "scissors paper stone mode";
+    return (
+      "Welcome, " +
+      userName +
+      ". Now key in 'scissors', 'paper', or 'stone' to play. Plus, try adding 'reverse' and see what happens'"
+    );
+  }
+
+  // If game mode has switched to 'scissors paper stone mode' ...
+  else if (gameMode == "scissors paper stone mode") {
+    var userGuess = input.toLowerCase();
+
+    // Eliminate NaN values for win rate in first round of game
+    if (playerWinCount == 0 || gameCount == 0) {
+      winRate = "0";
+    }
+
```

Comment:
link to above comment, as mentioned, can initialise  as var winRate =  ; to prevent NaN